### DISCUSSION TITLE
cannot compile c++ programs with mpicxx

### DISCUSSION THREAD

**Discussion by mari-125** [UIUC-PPL/charm#3383] [Q&A]
I am compiling NAMD source with MPI with this command:
    env MPICXX=mpicxx ./build charm++ mpi-linux-x86_64 --with-production

but I got: 
    cannot compile c++ programs with mpicxx 
   (check your charm++ version)
***Please find detailed output in tmp/charmconfig.out ***

gmake [1] *** No rule to make target 'conv-autoconfig.h', needed by 'xi-main.o' . Alto. 
gmake [1] Leaving directory '/data/users/msolis/ Namd/charm-6.10.2/mpi-linux-x86_64/tmp'
![WhatsApp Image 2021-06-04 at 18 00 39](https://user-images.githubusercontent.com/85375177/120873567-3b792d80-c568-11eb-9192-d8fd5050454d.jpeg)

please, could someone help me with this issue?

**Comment by rbuch** [comment]
The issue here is that your C++ compiler is misconfigured, it's trying to use `cc-mpicxx` when it should presumably be just `mpicxx`. In general, you shouldn't need to add an `env MPICXX=mpicxx` before the build command, Charm++ uses `mpicxx` as the MPI compiler by default, so try building without that part.

**Reply by mari-125** [reply]
Hello !, I ran this:

./build charm++ mpi-linux-x86_64 --with-production

and It generates the same result as above.

![WhatsApp Image 2021-06-04 at 22 51 06](https://user-images.githubusercontent.com/85375177/120879208-67f07200-c587-11eb-87d6-b7554751dd80.jpeg)

**Reply by matthiasdiener** [reply]
Can you paste the content of `tmp/charmconfig.out` here?

**Reply by mari-125** [reply]
Here it is:
1 Charm++/LIBS error checking is disabled
 2 AMPI error checking is enabled
 3 Statistics collection is disabled
 4 Charm tracing is disabled
 5 Charm tracing communication thread is disabled
 6 Task Queue is disabled
 7 Drone mode is disabled
 8 CharmDebug is disabled
 9 Charm record/replay is disabled
10 CCS is enabled
11 Charm control point is disabled
12 Charm LB user data is disabled
13 Lockless queue for PE local and node queue is disabled
14 Charm shrink expand is disabled
15 charm4py support is disabled
16 Setting load balancing timer type as 'double'
17 ### whether C++ compiler works
18 #include <stdio.h>
19 void foo(void) {
20         printf("Hello, world!\n");
21 }
22 mpicxx -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX -I../include -I. -c test.cpp -o test.o
23 ./configure: line 2838: mpicxx: command not found

**Reply by matthiasdiener** [reply]
It seems that you do not have MPI installed on your system. Can you try compiling an MPI application? In the meantime, you can try building a non-MPI version of Charm++, e.g. 

```
./build charm++ netlrts-linux-x86_64 --with-production
```

**Reply by matthiasdiener** [reply]
@mari-125 Were you able to fix this issue?
